### ISSUE TITLE
Is this maintained?

### ISSUE THREAD

**Issue by flexchar** [PrintNode/PrintNode-PHP#40]

**Comment by krizzdev** [comment]
We use this package since several years but are now very limited with some features. The most annoying thing is to set a given timeout value manually because it is hard coded int the curl request.

There is also a PR which isn't merged from last year: https://github.com/PrintNode/PrintNode-PHP/pull/32

**Comment by flexchar** [comment]
I contacted support as the package was not working for newly created account. Turns out API has changed and is only backwards compatible to old accounts. I have made account fixing the issue - #41. Hopefully they merge yours & mine soon.

**Comment by Nek-** [comment]
There's no version. Last commits break the API of the library. Bugs are not fixed. I believe that if somebody is motivated, this library should be fork.

**Comment by flexchar** [comment]
Ahh, they mentioned they gonna at least fix the critical bug. :((

**Comment by Nek-** [comment]
The work 11 days ago consists of removing the existing code, this will just break all our applications. Luckily they saved the master branch as "old master" so composer can still get the old (more recent ironically) code.

I'm not sure the situation is going any better here. 🙈

**Comment by shakethatweight-simon** [comment]
@Nek- I'm tempted to fork this, don't have much time to update the code, but with the branch renaming, I need old-master, but packagist hasn't been updated so I can't select "dev-old-master". By forking I could switch the branches back to the way they should be and start tagging version releases.

**Comment by Nek-** [comment]
@shakethatweight-simon you can specifically pick a commit, it works with composer and it's the best you can do I guess.

```json
{
    "require": {
		"printnode/printnode-php": "dev-master#960acabca3c0d976bbd92bf0d6490a1fe5df7496"
    }
}
```

I'm sorry to hear you have trouble with the package. I'm not affiliated at all with PrintNode, I can only wish you good luck.

**Comment by krizzdev** [comment]
Has anyone of you a maintained repository of the old-master ? Currently have the same issue because the namespaces changed in the new master branch

**Comment by shakethatweight-simon** [comment]
@krizzdev Looking at my composer.json it looks like we're using a fork I setup : https://github.com/shakethatweight-simon/PrintNode-PHP

**Comment by krizzdev** [comment]
Thanks @shakethatweight-simon for your quick answer.
I just found https://github.com/rawilk/laravel-printing this package here. I think I will build on that

**Comment by shakethatweight-simon** [comment]
@krizzdev Looks like https://github.com/rawilk/laravel-printing still uses this package too? https://randallwilk.dev/docs/laravel-printing/v2/installation#printnode

**Comment by krizzdev** [comment]
You're right. But with 187 stars I think it's more supported and maybe the abstraction is handled better. 
What I mean is that the chance is higher that laravel-printing will react quicker to fix/workaround an issue.

My whole application broke because printnode-php recommends using the dev-master branch and swapped it out in the last months.

**Comment by rawilk** [comment]
I'm the maintainer of laravel-printing, but I haven't really had a chance to do much work on it until now.

I personally use the `2.0@RC` version in my own projects and it seems to work okay, but the `Entity` class they have isn't compatible with `JsonSerializable`, which obviously is a problem. I'm going to try and find a workaround for this, but if anyone knows any working forks or alternatives for using the printnode api, that'd be great too.

**Comment by rawilk** [comment]
For anyone interested, I've just tagged `v3.0.0` of laravel-printing and it removes this package as a dependency entirely. I've written my own api wrapper for interacting with PrintNode's api.

**Comment by DjThossi** [comment]
What a shame for a paid product to give a shit about their SDKs. That's the best way of interaction with ANY API. And the easiest to onboard new customers. Oh and in the case of PrintNode, the easiest way to make money.

My recommendation to @PrintNodeLtd : open this project up and take someone from the community to support you with proper release cycles (tagging and such) and further PHP version support.

I'm working in a company which would even consider to fund this project IF it would be properly maintained. If more details required please let directly me know.
